scvi/model/_destvi.py

```diff
@@ -0,0 +1,343 @@
+            stdl = self._make_data_loader(
+                adata=self.adata, indices=indices, batch_size=batch_size
+            )
+            gamma_ = []
+            for tensors in stdl:
+                generative_inputs = self.module._get_generative_input(tensors, None)
+                gamma_local = self.module.get_gamma(x=generative_inputs["x"])
+                gamma_ += [gamma_local.cpu()]
+            data = np.array(torch.cat(gamma_, dim=-1))
+            if indices is not None:
+                index_names = index_names[indices]
+        else:
+            if indices is not None:
+                logger.info(
+                    "No amortization for latent values, ignoring adata and returning results for the full data"
+                )
+            data = self.module.get_gamma()
+
+        data = np.transpose(data, (2, 0, 1))
+        res = {}
+        for i, ct in enumerate(self.cell_type_mapping):
+            res[ct] = pd.DataFrame(
+                data=data[:, :, i], columns=column_names, index=index_names
+            )
+
+        if return_numpy:
+            return data
+
+        return res
+
+    def get_scale_for_ct(
+        self,
+        indices: Sequence[int],
+        label: str,
+        batch_size: Optional[int] = None,
+    ) -> pd.DataFrame:
+        r"""
+        Return the scaled parameter of the NB for every spot in queried cell types.
+
+        Parameters
+        ----------
+        indices
+            Indices of cells in self.adata to use. If `None`, all cells are used.
+        label
+            cell type of interest
+        batch_size
+            Minibatch size for data loading into model. Defaults to `scvi.settings.batch_size`.
+
+        Returns
+        -------
+        Pandas dataframe of gene_expression
+        """
+        if self.is_trained_ is False:
+            logger.warning(
+                "Trying to query inferred values from an untrained model. Please train the model first."
+            )
+
+        if label not in self.cell_type_mapping:
+            raise ValueError("Unknown cell type")
+        y = np.where(label == self.cell_type_mapping)[0][0]
+
+        stdl = self._make_data_loader(
+            self.adata, indices=indices, batch_size=batch_size
+        )
+        scale = []
+        for tensors in stdl:
+            generative_inputs = self.module._get_generative_input(tensors, None)
+            x, ind_x = generative_inputs["x"], generative_inputs["ind_x"]
+            px_scale = self.module.get_ct_specific_expression(x, ind_x, y)
+            scale += [px_scale.cpu()]
+        return np.array(torch.cat(scale))
+
+    def train(
+        self,
+        max_epochs: int = 400,
+        lr: float = 0.01,
+        use_gpu: Optional[Union[str, int, bool]] = None,
+        train_size: float = 1.0,
+        validation_size: Optional[float] = None,
+        batch_size: int = 128,
+        plan_kwargs: Optional[dict] = None,
```

Comment:
you could expose the early stopping args here if it makes sense for this model

Reply:
this doesn't here